modules/update_folders.nf

```diff
@@ -7,11 +7,10 @@ process UPDATE_FOLDERS {
     input:
     val submission_id
     val project_name
-    path predictions_file
-    path docker_log_file
+    tuple path(predictions_file), path(docker_log_file)
 
     output:
-    val "ready"
+    tuple val(submission_id), path(predictions_file), val("status"), path("output_annotation*.json")
```

Comment:
Rather than using a separate JSON file here for these annotations, would it work to just start the `results.json` file here so that we are using one common file for all annotations/results throughout the workflow?

Reply:
I think this would work, with the caveat that the annotations would be overwritten in the validation step: The PEGS challenge organizers' [validation script](https://github.com/Sage-Bionetworks-Challenges/pegs-evaluation/blob/aa69ecac04ea1a4dadf45990f989876940c472e9/validate.py#L140) for example writes into `results.json` with the assumption that it hasn't been made before. That line there will overwrite the contents in the `results.json` with just the validation results.

But by this point, the annotations have already been added to the submission on Synapse, so it wouldn't matter, except from a troubleshooting standpoint I guess?

Reply:
Ah I see. I don't love using the same file and overwriting it for the reason you outlined. I think we just keep an eye on this for the future as you say. Thanks!